recorder/src/main/cpp/profiler.cc

```diff
@@ -25,42 +26,50 @@ void Profiler::handle(int signum, siginfo_t *info, void *context) {
     ThreadBucket *thread_info = nullptr;
     PerfCtx::ThreadTracker* ctx_tracker = nullptr;
     auto current_sampling_attempt = sampling_attempts.fetch_add(1, std::memory_order_relaxed);
+    bool default_ctx = false;
     if (jniEnv != nullptr) {
         thread_info = thread_map.get(jniEnv);
         bool do_record = true;
         if (thread_info != nullptr) {//TODO: increment a counter here to monitor freq of this, it could be GC thd or compiler-broker etc
             ctx_tracker = &(thread_info->ctx_tracker);
-            do_record = ctx_tracker->in_ctx() ? ctx_tracker->should_record() : get_prob_pct().on(current_sampling_attempt, noctx_cov_pct);
+            if (ctx_tracker->in_ctx()) {
+                do_record = ctx_tracker->should_record();
+            } else {
+                do_record = get_prob_pct().on(current_sampling_attempt, noctx_cov_pct);
+                default_ctx = true;
+            }
         }
         if (! do_record) {
             return;
         }
     }
 
-    STATIC_ARRAY(frames, JVMPI_CallFrame, capture_stack_depth(), MAX_FRAMES_TO_CAPTURE);
-
-    JVMPI_CallTrace trace;
-    trace.frames = frames;
-    
-    if (jniEnv == NULL) {
-        IsGCActiveType is_gc_active = Asgct::GetIsGCActive();
```

Comment:
The C++ class-name is usually indicative enough. Eg. we have a test that allocates too much (CpuSamplingTest#should_Report_Alloc_Burn) and it has ...ParallelScavengeHeap::mem_allocate... on the backtrace for 90% of the samples. Some are due to GC, others due to allocation.